src/allmydata/mutable/publish.py

```diff
@@ -704,7 +704,7 @@ def _encode_segment(self, segnum):
         self.log("Pushing segment %d of %d" % (segnum + 1, self.num_segments))
         data = self.data.read(segsize)
         # XXX: This is dumb. Why return a list?
-        data = "".join(data)
+        data = b"".join(data)
```

Comment:
Is that XXX comment irrelevant? Not really part of this PR but ... it's not returning a list :)

Reply:
Hm. Maybe it's commenting on `self.data.read()`?